packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_submit_ext_htlc.test.ts

```diff
@@ -0,0 +1,208 @@
+        height: expect.any(BigNumber)
+      }
+    )
+  })
+
+  it('should return an error when ICXSubmitEXTHTLC.amount is negative', async () => {
+    // ICX order creation, make offer and submit DFC HTLC
+    const createOrder = {
+      chainTo: 'BTC',
+      ownerAddress: testing.icxorderbook.accountDFI,
+      receivePubkey: '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941',
+      amountFrom: new BigNumber(15),
+      orderPrice: new BigNumber(0.01)
+    }
+    const { createOrderTxId } = await testing.icxorderbook.createDFISellOrder(createOrder)
+
+    const makeOffer = {
+      orderTx: createOrderTxId,
+      amount: new BigNumber(0.10),
+      ownerAddress: testing.icxorderbook.accountBTC
+    }
+    const { makeOfferTxId } = await testing.icxorderbook.createDFIBuyOffer(makeOffer)
+
+    const DFCHTLC = {
+      offerTx: makeOfferTxId,
+      amount: new BigNumber(10),
+      hash: '957fc0fd643f605b2938e0631a61529fd70bd35b2162a21d978c41e5241a5220',
+      timeout: 1440
+    }
+    await testing.icxorderbook.createDFCHTLCForDFIBuyOffer(DFCHTLC)
+
+    // submit EXT HTLC
+    const script = await providers.elliptic.script()
+    const submitEXTHTLC: ICXSubmitEXTHTLC = {
+      offerTx: makeOfferTxId,
+      amount: new BigNumber(-0.10),
+      hash: '957fc0fd643f605b2938e0631a61529fd70bd35b2162a21d978c41e5241a5220',
+      htlcScriptAddress: '13sJQ9wBWh8ssihHUgAaCmNWJbBAG5Hr9N',
+      ownerPubkey: '036494e7c9467c8c7ff3bf29e841907fb0fa24241866569944ea422479ec0e6252',
+      timeout: 24
+    }
+    await expect(builder.icxorderbook.submitDFCHTLC(submitEXTHTLC, script)).rejects.toThrow('The value of "value" is out of range. It must be >= 0 and <= 4294967295. Received -10000000')
+  })
+
+  it('should return an error when ICXSubmitEXTHTLC.timeout is negative', async () => {
+    // ICX order creation, make offer and submit DFC HTLC
+    const createOrder = {
+      chainTo: 'BTC',
+      ownerAddress: testing.icxorderbook.accountDFI,
+      receivePubkey: '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941',
+      amountFrom: new BigNumber(15),
+      orderPrice: new BigNumber(0.01)
+    }
+    const { createOrderTxId } = await testing.icxorderbook.createDFISellOrder(createOrder)
+
+    const makeOffer = {
+      orderTx: createOrderTxId,
+      amount: new BigNumber(0.10),
+      ownerAddress: testing.icxorderbook.accountBTC
+    }
+    const { makeOfferTxId } = await testing.icxorderbook.createDFIBuyOffer(makeOffer)
+
+    const DFCHTLC = {
+      offerTx: makeOfferTxId,
+      amount: new BigNumber(10),
+      hash: '957fc0fd643f605b2938e0631a61529fd70bd35b2162a21d978c41e5241a5220',
+      timeout: 1440
+    }
+    await testing.icxorderbook.createDFCHTLCForDFIBuyOffer(DFCHTLC)
+
+    // submit EXT HTLC
+    const script = await providers.elliptic.script()
+    const submitEXTHTLC: ICXSubmitEXTHTLC = {
+      offerTx: makeOfferTxId,
+      amount: new BigNumber(0.10),
```

Comment:
wondering what happened if set diff amount
```suggestion
      amount: new BigNumber(0.10),
      amount: new BigNumber(0.05),
```

Reply:
https://github.com/DeFiCh/jellyfish/blob/a980b3b2aa15c6f13f8380aa366333b5d11b1133/packages/jellyfish-api-core/__tests__/category/icxorderbook/submitExtHTLC.test.ts#L329